src/cocotb/task.py

```diff
@@ -41,20 +61,10 @@ class Task(Generic[ResultType]):
         Use :meth:`result`, :meth:`done`, and :meth:`done` methods instead, respectively.
     """
 
-    class _State(DocEnum):
-        """State of a Task."""
-
-        UNSTARTED = (auto(), "Task created, but never run and not scheduled")
-        SCHEDULED = (auto(), "Task in Scheduler queue to run soon")
-        PENDING = (auto(), "Task waiting for Trigger to fire")
-        RUNNING = (auto(), "Task is currently running")
-        FINISHED = (auto(), "Task has finished with a value or Exception")
-        CANCELLED = (auto(), "Task was cancelled before it finished")
-
-    _name: str = "Task"  # class name of schedulable task
     _id_count = 0  # used by the scheduler for debug
 
-    def __init__(self, inst):
+    def __init__(self, inst: Coroutine[cocotb.triggers.Trigger, None, Any]) -> None:
+        self._native_coroutine = False
```

Comment:
Get rid of this variable, switch to a `inspect.iscoroutine` check instead. Remove the check in `_get_coro_stack`. It's private and we guard it's usage in `__repr__` so we don't have to worry about doing redundant checks.